client/state/happychat/test/selectors.js

```diff
@@ -58,6 +70,59 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( '#canUserSendMessages', () => {
+		const messagingDisabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ABANDONED,
+			HAPPYCHAT_CHAT_STATUS_BLOCKED,
+			HAPPYCHAT_CHAT_STATUS_DEFAULT,
+			HAPPYCHAT_CHAT_STATUS_MISSED,
+			HAPPYCHAT_CHAT_STATUS_PENDING,
+		];
+		const messagingEnabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ASSIGNED,
+			HAPPYCHAT_CHAT_STATUS_ASSIGNING,
+			HAPPYCHAT_CHAT_STATUS_CLOSED,
+			HAPPYCHAT_CHAT_STATUS_NEW,
+		];
+
+		it( 'should return false if Happychat is unavailable', () => {
+			const state = {
+				happychat: {
+					connectionStatus: 'uninitialized',
+					isAvailable: false,
+					chatStatus: HAPPYCHAT_CHAT_STATUS_NEW
+				}
+			};
+			expect( canUserSendMessages( state ) ).to.be.false;
+		} );
+
+		it( "should return false if Happychat is available but the chat status doesn't allow messaging", () => {
+			messagingDisabledChatStatuses.forEach( status => {
+				const state = {
```

Comment:
Seems tedious but I try to `deepfreeze` objects like this to make sure the selector doesn't accidentally modify state.